darshan-test/c_mpi_progs/runtime_prog_issue_730.c

```diff
@@ -0,0 +1,23 @@
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <mpi.h>
+#include <stdlib.h>
+
+
+int main(int argc, char **argv)
+{
+   int rank;
+   int fd;
+   char c;
+   
+   MPI_Init(&argc, &argv);
+   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+   fd = open("test.txt", O_RDONLY);
+   read(fd, &c, 1);
+   for(int i = 0; i < 5000; i++) {
+       read(fd, &c, 0);
+   }
+   MPI_Finalize();
+}
```

Comment:
@carns @shanedsnyder Here is an example of a "C-based" regression test in the end-to-end functional testing infrastructure. At the time of writing, this passes instead of causing the issue Phil suggested here:
https://github.com/darshan-hpc/darshan/pull/737#issuecomment-1126148539

I'm open to suggestions, or feel free to push in to this PR if you think this can be made to repro the `0/0`.